node/node.go

```diff
@@ -927,8 +927,10 @@ func (node *AlgorandFullNode) InstallParticipationKey(partKeyBinary []byte) (acc
 	}
 
 	// Tell the AccountManager about the Participation (dupes don't matter) so we ignore the return value
-	_ = node.accountManager.AddParticipation(partkey)
-
+	added := node.accountManager.AddParticipation(partkey)
+	if !added {
+		return account.ParticipationID{}, fmt.Errorf("ParticipationRegistry: cannot register duplicate participation key")
```

Comment:
It looks like this error message might be wrong. Are duplicate participation keys the only reason AddParticipation would fail?

Reply:
Looking at the function, yes it would be the only reason it seems.

```
	_, alreadyPresent := manager.partKeys[partkeyID]
	if alreadyPresent {
		return false
	}

       .....
	return true

```